cluster/addons/iso-client/opaque/opaque.go

```diff
@@ -60,80 +37,67 @@ func getNode() (string, error) {
 	return os.Hostname()
 }
 
-// path for opaque resources
-func (opaque *OpaqueIntegerResourceAdvertiser) generateOpaqueResourcePath() string {
-	return fmt.Sprintf("/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-%s", opaque.Name)
+// Escape forward slashes in the resource name per the JSON Pointer spec.
+// See https://tools.ietf.org/html/rfc6901#section-3
+func escapeResourcePath(resName api.ResourceName) string {
+	return strings.Replace(string(resName), "/", "~1", -1)
 }
 
-// generate url for adding or removing opaque resources
-func (opaque *OpaqueIntegerResourceAdvertiser) generateOpaqueResourceUrl() string {
-	return fmt.Sprintf("%s/api/v1/nodes/%s/status", opaque.Config.Host, opaque.Node)
+// genereate OIR path for patch operation
+func opaqueResourcePath(name string) string {
+	return fmt.Sprintf("/status/capacity/%s", escapeResourcePath(api.OpaqueIntResourceName(name)))
 }
 
 // Getting config for accesing apiserver, assuming pod ir run within the cluster
 func getClientConfig() (*restclient.Config, error) {
 	return restclient.InClusterConfig()
 }
 
+func getApiServer() (string, error) {
+	config, err := getClientConfig()
+	if err != nil {
+		return "", err
+	}
+	return config.Host, nil
+}
+
 // prepare PATCH request for adding/removing opaque resources
 func prepareRequest(body []byte, url string) (*http.Request, error) {
 	return http.NewRequest(http.MethodPatch, url, bytes.NewBuffer(body))
 }
 
-//Setting application/json-patch+json header
-func setPatchHeader(req *http.Request) {
-	req.Header.Set("Content-Type", "application/json-patch+json")
-}
-
-func (opaque *OpaqueIntegerResourceAdvertiser) createHttpClient() (*http.Client, error) {
-	// TODO: test it against insecure apiserver
-	transportConfig, err := opaque.Config.TransportConfig()
-	if err != nil {
-		return nil, err
-	}
-	transport, err := transport.New(transportConfig)
-	if err != nil {
-		return nil, err
-	}
-	return &http.Client{Transport: transport}, nil
-}
-
-func (opaque *OpaqueIntegerResourceAdvertiser) AdvertiseOpaqueResource() error {
-	return opaque.makeRequest("add")
+func AdvertiseOpaqueResource(name string, value int) error {
+	return makeRequest("add", name, value)
 }
 
-func (opaque *OpaqueIntegerResourceAdvertiser) RemoveOpaqueResource() error {
-	return opaque.makeRequest("remove")
+func RemoveOpaqueResource(name string) error {
+	return makeRequest("remove", name, 1)
```

Comment:
This is kind of awkward. The third parameter is confusing and also yields a confusing request body. Could we change `makeRequest` to take a patch operation struct instead? Then combined with omitempty for the JSON serializer we can easily have remove operations without a value field.